lib/SIL/IR/SILFunctionType.cpp

```diff
@@ -2708,6 +2708,25 @@ class CXXMethodConventions : public CFunctionTypeConventions {
       return ParameterConvention::Indirect_In_Guaranteed;
     return ParameterConvention::Indirect_Inout;
   }
+  ResultConvention getResult(const TypeLowering &resultTL) const override {
+    if (dyn_cast<clang::CXXConstructorDecl>(TheDecl)) {
```

Comment:
random drive by, but please use isa<> in a boolean context instead of dyn_cast.

There are also a number of 80 column violations that need to be fixed on comments and other things in this patch.